core/celery/tasks/generate_inventory_and_hold_reports.py

```diff
@@ -0,0 +1,292 @@
+from __future__ import annotations
+
+import csv
+import os
+import tempfile
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+from celery import shared_task
+from sqlalchemy import not_, select, text
+from sqlalchemy.orm import Session, sessionmaker
+
+from api.integration.registry.license_providers import LicenseProvidersRegistry
+from core.celery.job import Job
+from core.celery.task import Task
+from core.integration.goals import Goals
+from core.model import (
+    IntegrationConfiguration,
+    IntegrationLibraryConfiguration,
+    Library,
+    get_one,
+)
+from core.opds_import import OPDSImporterSettings
+from core.service.celery.celery import QueueNames
+from core.service.container import Services, container_instance
+
+
+class GenerateInventoryAndHoldsReportsJob(Job):
+    _delete_attachments = True
```

Comment:
Minor: Might be better to make this a instance variable, and pass it into the constructor instead.